File: test/fixtures/fake-stdio-adapter.mjs
if (process.env.FAKE_ADAPTER_MODE === 'early-exit') process.exit(0)
let raw = ''
for await (const chunk of process.stdin) raw += chunk
const request = JSON.parse(raw)
const mode = process.env.FAKE_ADAPTER_MODE
if (mode === 'malformed') process.stdout.write('not json')
else if (mode === 'failed') process.stdout.write(JSON.stringify({ protocolVersion: 'relay.adapter/v1', status: 'failed', error: 'fake failure' }))
else if (mode === 'oversized') process.stdout.write(JSON.stringify({ protocolVersion: 'relay.adapter/v1', status: 'completed', output: 'x'.repeat(10000) }))
else process.stdout.write(JSON.stringify({ protocolVersion: 'relay.adapter/v1', status: 'completed', output: `${request.task.sessionId}:${request.task.input}` }))

File: test/fixtures/fake_stdio_adapter.py
#!/usr/bin/env python3
"""Fake relay.adapter/v1 stdio adapter used by the test suite."""
from __future__ import annotations

import json
import os
import sys


def main():
    mode = os.environ.get("FAKE_ADAPTER_MODE")
    if mode == "early-exit":
        return
    request = json.loads(sys.stdin.read())
    task = request["task"]
    if mode == "malformed":
        sys.stdout.write("not json")
    elif mode == "failed":
        sys.stdout.write(json.dumps(
            {"protocolVersion": "relay.adapter/v1", "status": "failed", "error": "fake failure"}))
    elif mode == "oversized":
        sys.stdout.write(json.dumps(
            {"protocolVersion": "relay.adapter/v1", "status": "completed", "output": "x" * 10000}))
    else:
        sys.stdout.write(json.dumps(
            {"protocolVersion": "relay.adapter/v1", "status": "completed",
             "output": f"{task['sessionId']}:{task['input']}"}))


if __name__ == "__main__":
    main()
